### DISCUSSION TITLE
Is `get valueWatcher => ref.watch(valueProvider)` inadvasied?

### DISCUSSION THREAD

**Discussion by spauldhaliwal** [rrousselGit/riverpod#3590] [Q&A] [answered]
I'm wondering if there are any downsides or potential bugs associated with defining my watchers like the following:

`String? value;`

`String get _valueWatcher => ref.watch(valueProvider);`

```
@override
  Widget build(BuildContext context) {
    value = _valueWatcher;
    return MyWidget();
  }
```
  
I prefer this syntax as it allows me to more easily document the value parameters and "watcher" parameters without cluttering the build function. So far I haven't noticed any bugs or downsides, but I'm wondering if this might lead to hard-to-find bugs down the road. From what I understand, Riverpod is a little finicky about what gets placed in the build functions or not.

Thanks!

**Comment by snapsl** [comment, marked as answer]
Only works in StatefulConsumerWidget, not in ConsumerWidgets. People trying to read your code might get confused. Other than that, you are fine.
